pkg/network/tracer/tracer_test.go

```diff
@@ -1017,12 +1017,13 @@ func testDNSStats(t *testing.T, tr *Tracer, domain string, success, failure, tim
 	queryMsg.RecursionDesired = true
 
 	dnsClient := new(dns.Client)
-	dnsConn, err := dnsClient.Dial(dnsServerAddr.String())
-	require.NoError(t, err)
-	defer dnsConn.Close()
-	dnsClientAddr := dnsConn.LocalAddr().(*net.UDPAddr)
+	var dnsClientAddr *net.UDPAddr
 	require.Eventually(t, func() bool {
+		dnsConn, err := dnsClient.Dial(dnsServerAddr.String())
+		require.NoError(t, err)
```

Comment:
Does this fail the test? There is also a version of Eventually that passes in a context. https://pkg.go.dev/github.com/stretchr/testify/assert#EventuallyWithT

Reply:
this passes locally, I've been running each branch 250 times to ensure it's no longer flaky:
```
--- PASS: TestTracerSuite (11.15s)
    --- PASS: TestTracerSuite/runtime_compiled (11.15s)
        --- PASS: TestTracerSuite/runtime_compiled/TestDNSStats (7.03s)
            --- PASS: TestTracerSuite/runtime_compiled/TestDNSStats/valid_domain (0.61s)
            --- PASS: TestTracerSuite/runtime_compiled/TestDNSStats/invalid_domain (0.63s)
            --- PASS: TestTracerSuite/runtime_compiled/TestDNSStats/timeout (2.61s)
        --- PASS: TestTracerSuite/runtime_compiled/TestDNSStatsWithNAT (4.11s)
PASS
ok  	github.com/DataDog/datadog-agent/pkg/network/tracer	2793.268s
```

Reply:
I meant using `require.Error` there inside `require.Eventually` would fail the test if it trips?

Reply:
I guess it doesn't matter since you switched to context.